src/lib/queue.js

```diff
@@ -61,6 +61,19 @@ const {
 
 export { generateRedisConfig };
 
+export async function increaseListeners(queue, num = 1) {
+  const redisClient = queue.client;
+  const maxListeners = redisClient.getMaxListeners();
+  const currentCounts = queue.eventNames().reduce((counts, eventName) => ({
+    ...counts,
+    [eventName]: queue.listenerCount(eventName),
+  }), {});
+  const totalCount = Object.values(currentCounts).reduce((acc, count) => acc + count, 0);
+  if (totalCount + num > maxListeners) {
+    redisClient.setMaxListeners(Math.max(totalCount + num, maxListeners));
```

Comment:
Does it make sense to just increase the max listeners to an arbitrary yet realistic number instead of just scaling this infinitely? This would hide exactly the kind of mistake that max listeners is supposed to highlight

Reply:
@GarrettEHill can you comment on this?

Reply:
> MaxListenersExceededWarning: Possible EventEmitter memory leak detected. 11 ready listeners added to [Redis]. Use emitter.setMaxListeners() to increase limit
 Without this we are getting this warning. So we need something to exstend the value

Reply:
We would still likely need something to check for runnaway listeners though

Reply:
The warning is not saying redis can't handle anymore listeners, the warning is saying "hey we noticed you are adding a lot of listeners, is something wrong bud?" 

So it doesn't make sense to scale that number up forever, is what I'm saying, since all it would do is hide a potentially valuable warning.

Reply:
I suppose that this would increase the max listeners we expect, and any number over that would mean a memory leak or something else fishy. 

I wonder if we might want to include a maximum overall number before setting the max listeners.

Reply:
Added cap.

Reply:
See my comment elsewhere, I put the Math.min in the wrong place.